shap/maskers/_tabular.py

```diff
@@ -61,6 +63,16 @@ def __init__(self, data, max_samples=100, clustering=None):
         self.clustering = clustering
         self.max_samples = max_samples
 
+        # prepare by fitting sklearn imputer
```

Comment:
I would rather see this in the impute than in the generic tabular class

Reply:
I'm not sure I understand. Do you suggest overwriting the `__call__` method for `Impute`? I agree that this shaping check could be moved to the `LinearImpute` and `Impute` classes (sklearn imputers expect 2-dimensional arrays, so I kept that consistent in `LinearImpute`). However, some form of `self.impute.fit()` and `self.impute.transform()` still need to be called during the main algorithm as written.